src/exercise/04.js

```diff
@@ -5,21 +5,26 @@ import * as React from 'react'
 
 function Board() {
   // 🐨 squares is the state for this component. Add useState for squares
-  const squares = Array(9).fill(null)
+  const [squares, setSquares] = React.useState(Array(9).fill(null));
 
   // 🐨 We'll need the following bits of derived state:
   // - nextValue ('X' or 'O')
   // - winner ('X', 'O', or null)
   // - status (`Winner: ${winner}`, `Scratch: Cat's game`, or `Next player: ${nextValue}`)
   // 💰 I've written the calculations for you! So you can use my utilities
   // below to create these variables
+  const nextValue = calculateNextValue(squares);
+  const winner = calculateWinner(squares);
+  const status = calculateStatus(winner, squares, nextValue);
```

Comment:
I'm not sure if this is totally necessary, but in my implementation I wrapped these computations in a `useEffect` hook and set the dependencies array to `[squares]` and set these values as state variables rather than `const`s.

Your implementation may actually be better here because changing state triggers a re-render of the component, so I'm probably re-rendering my component more than I need to. If this works reliably, then 👍